File: windows/PomodoroTimerApp/PomodoroTimers/States/PausedState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Timers;

namespace PomodoroTimerApp.PomodoroTimers.States
{
    internal class PausedState : State
    {
        public PausedState(PomodoroTimer pomodoroTimer) : base(pomodoroTimer)
        {
        }

        public override bool ActivityPause()
        {
            return false;
        }
        public override bool ActivityResume()
        {
            _timer.Resume();
            _timer.ChangeState(new RunningState(_timer));
            return true;
        }

        public override void Completed()
        {
            // Do nothing
        }


        public override void StartPauseResume()
        {
            _timer.Resume();
            _timer.ChangeState(new RunningState(_timer));
        }

        public override void Stop()
        {
            _timer.Stop();
            _timer.ChangeState(new ReadyState(_timer));
        }
    }
}

File: Sources/PomodoroCore/PomodoroTimers/States/PausedState.swift
import Foundation

struct PausedState: TimerState {
    let kind: TimerStateKind = .paused

    func startPauseResume(_ timer: PomodoroTimer) {
        timer.resume()
        timer.change(to: RunningState())
    }

    func stop(_ timer: PomodoroTimer) {
        timer.stop()
        timer.change(to: ReadyState())
    }

    func completed(_ timer: PomodoroTimer) {}
    func activityPause(_ timer: PomodoroTimer) -> Bool { false }

    func activityResume(_ timer: PomodoroTimer) -> Bool {
        timer.resume()
        timer.change(to: RunningState())
        return true
    }
}
